src/webgpu/api/operation/buffers/map.spec.ts

```diff
@@ -125,7 +126,7 @@ map-read and check the read-back result.`
     const mapRegion = getRegionForMap(size, [rangeOffset, rangeSize], t.params);
     await buffer.mapAsync(GPUMapMode.READ, ...mapRegion);
     const actual = new Uint8Array(buffer.getMappedRange(...range));
-    t.expectBuffer(actual, new Uint8Array(expected.buffer));
+    t.expectOK(checkElementsEqual(actual, new Uint8Array(expected.buffer)));
```

Comment:
nit: `this.expectGPUBufferValuesEqual` instead? In general it feels very few places in the code should need to use `expectOk`

Reply:
This isn't checking a GPUBuffer though, so I'd be adding a new `expectElementsEqual` to the fixture for just a few cases.

I thought about adding an option to make expectGPUBufferValuesEqual readback using mapAsync. I was thinking it wouldn't be able to be `eventual` but actually it could consume the input GPUBuffer so it wouldn't have to be.

Reply:
I think I found a preexisting bug in expectGPUBufferValuesPassCheck (`begin` and `end` are byte offsets but it treats them as array offsets), so I'm going to try to fix that separately then do this.

Reply:
ok

Reply:
I didn't end up fixing this one because it can't be folded with expectGPUBufferValuesEqual because it is testing the map itself, but it would also add a one-off expectElementsEqual that isn't used anywhere else. I think this is okay. Also I may adopt the expectOK(checkX()) pattern in the future if if it's handy, as it's more composable.

Reply:
Ack